src/key-value-engine/B-Tree-vs-Log-Structured-Merge-Tree.md

```diff
@@ -0,0 +1,130 @@
+# B-Tree vs Log-Structured Merge-Tree
+
+[B-tree](https://en.wikipedia.org/wiki/B-tree) and [Log-Structured Merge-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree) (LSM-tree) are the two most widely used data structures for data-intensive applications to organize and store data. However, each of them has its own advantages and disadvantages. This article aims to use the quantitative approaches to compare these two data structures. 
+
+## Metrics
+
+In general, there are three most important metrics to measure the performance of a data structure, which include write amplification, read amplification, and space amplification. This section aims to describe these metrics. 
+
+### Write Amplification
+
+`Write amplification` is the ratio of the amount of data written to the storage device versus the amount of data written to the database. 
+
+For example, if you are writing 10 MB to the database and you observe 30 MB disk write rate, your write amplification is 3.
+
+Flash memory and solid-state drives (SSDs) can be written to only a finite number of times, so that write amplification will decrease the flash lifetime.
```

Comment:
`so that write amplification will decrease the flash lifetime.` Might read better as `so write amplification will decrease the drive lifetime`.